packages/analytics/src/integrations/Omnitracking/omnitracking-helper.ts

```diff
@@ -521,6 +521,27 @@ export const getProductLineItems = (data: EventData<TrackTypesValues>) => {
   return undefined;
 };
 
+/**
+ * Get the first product id from line items omnitracking parameter.
+ *
+ * @param data - The event tracking data.
+ *
+ * @returns The first product id from line items parameter.
+ */
+export const getProductIdFromLineItems = (
+  data: EventData<TrackTypesValues>,
+) => {
+  const lineItems = getProductLineItems(data);
+
+  if (lineItems) {
+    const products = JSON.parse(lineItems);
+
+    return products[0]['productId'];
```

Comment:
The code should not assume that lineItems is an array. It should add a call to `Array.isArray` for the `products` variable and also check if its length is greater than 0.

Reply:
Fixed 👍🏼